doc/source/io.rst

```diff
@@ -4574,6 +4574,8 @@ Several caveats.
 * Categorical dtypes can be serialized to parquet, but will de-serialize as ``object`` dtype.
 * Non supported types include ``Period`` and actual Python object types. These will raise a helpful error message
   on an attempt at serialization.
+* ``partition_cols`` will be used for partitioning the dataset, where the dataset will be written to multiple
```

Comment:
The rest of the items in this lists feel more like limitations of pandas / these engines. Requiring that `path` be a directory when `partition_cols` is set doesn't seem to fit here.

I think this is important / different enough to deserve a new small section below "Handling Indexes", with

1. A description of what `partition_cols` requires (list of column names, directory for file path)
2. A description of why you might want to use partition_cols
3. A small example.

Reply:
done